blockchain/blockchain.go

```diff
@@ -1012,7 +1013,7 @@ func (bc *BlockChain) writeStateTrie(block *types.Block, state *state.StateDB) e
 
 	// If we're running an archive node, always flush
 	if bc.isArchiveMode() {
-		if err := trieDB.Commit(root, false); err != nil {
+		if err := trieDB.Commit(root, false, block.NumberU64()); err != nil {
```

Comment:
I think we should not perform blockchain data archiving if "archive" mode is on.
What happens if we pass the current block number to trieDB.Commit()? Does it perform new db creation?

Reply:
@kjhman21 Yes, it will. Instead, we can put `NoDataArchivingPreparation` to indicate that no data archiving feature for an archiving node.

Reply:
@ehnuje So, what is your plan? Will this be changed to `NoDataArchivingPreparation` or something else?

Reply:
@kjhman21 I didn't consider the situation where the data archiving feature is enabled on the archiving node. (BTW, changing the names seems to be needed 😅 ) Let me update the PR.